### ISSUE TITLE
Reallocate Job Priorities

### ISSUE THREAD

**Issue by cottsay** [ros2/ros_buildfarm_config#49]
The Jenkins job priorities have previously been allocated such that certain job types are prioritized above other job types, and within each job type, more recent ROS distributions are prioritized above older ones.

We need to increase the spacing between these job types to avoid priority collisions when we add jobs for Foxy.

Additionally, we should consider adding a group for the manual CI jobs (overlay) which would be prioritized above the nightly jobs.

Current allocations:
<details>

```
45
  jenkins_pull_request_job_priority::eloquent/source-build.yaml
46
  jenkins_pull_request_job_priority::dashing/source-build.yaml
47
  jenkins_pull_request_job_priority::crystal/source-build.yaml
48
  jenkins_job_priority::eloquent/ci-nightly-connext.yaml
  jenkins_job_priority::eloquent/ci-nightly-cyclonedds.yaml
  jenkins_job_priority::eloquent/ci-nightly-debug.yaml
  jenkins_job_priority::eloquent/ci-nightly-extra-rmw-release.yaml
  jenkins_job_priority::eloquent/ci-nightly-fastrtps-dynamic.yaml
  jenkins_job_priority::eloquent/ci-nightly-fastrtps.yaml
  jenkins_job_priority::eloquent/ci-nightly-navigation2-prerequisites.yaml
  jenkins_job_priority::eloquent/ci-nightly-navigation2.yaml
  jenkins_job_priority::eloquent/ci-nightly-opensplice.yaml
  jenkins_job_priority::eloquent/ci-nightly-release.yaml
  jenkins_job_priority::eloquent/ci-overlay.yaml
49
  jenkins_job_priority::dashing/ci-nightly-connext.yaml
  jenkins_job_priority::dashing/ci-nightly-cyclonedds.yaml
  jenkins_job_priority::dashing/ci-nightly-debug.yaml
  jenkins_job_priority::dashing/ci-nightly-extra-rmw-release.yaml
  jenkins_job_priority::dashing/ci-nightly-fastrtps-dynamic.yaml
  jenkins_job_priority::dashing/ci-nightly-fastrtps.yaml
  jenkins_job_priority::dashing/ci-nightly-opensplice.yaml
  jenkins_job_priority::dashing/ci-nightly-release.yaml
  jenkins_job_priority::dashing/ci-overlay.yaml
50
  jenkins_job_priority::crystal/ci-nightly.yaml
  jenkins_job_priority::crystal/ci-overlay.yaml
55
  jenkins_commit_job_priority::eloquent/source-build.yaml
56
  jenkins_commit_job_priority::dashing/source-build.yaml
57
  jenkins_commit_job_priority::crystal/source-build.yaml
75
  jenkins_source_job_priority::eloquent/release-bionic-arm64-build.yaml
  jenkins_source_job_priority::eloquent/release-bionic-armhf-build.yaml
  jenkins_source_job_priority::eloquent/release-build.yaml
76
  jenkins_source_job_priority::dashing/release-bionic-arm64-build.yaml
  jenkins_source_job_priority::dashing/release-bionic-armhf-build.yaml
  jenkins_source_job_priority::dashing/release-build.yaml
77
  jenkins_source_job_priority::crystal/release-bionic-arm64-build.yaml
  jenkins_source_job_priority::crystal/release-build.yaml
79
  jenkins_binary_job_priority::eloquent/release-bionic-arm64-build.yaml
  jenkins_binary_job_priority::eloquent/release-build.yaml
80
  jenkins_binary_job_priority::dashing/release-bionic-arm64-build.yaml
  jenkins_binary_job_priority::dashing/release-build.yaml
81
  jenkins_binary_job_priority::crystal/release-bionic-arm64-build.yaml
  jenkins_binary_job_priority::crystal/release-build.yaml
99
  jenkins_binary_job_priority::dashing/release-bionic-armhf-build.yaml
  jenkins_binary_job_priority::eloquent/release-bionic-armhf-build.yaml
```

</details>

**Comment by dirk-thomas** [comment]
We have the following job categories in priority order:

* PR jobs
* nightly CI jobs
* dev jobs
* source deb jobs
* binary deb jobs
  * amd64 and aarch64
  * armhf

The overall order of these categories makes sense imo - a very brief rational summary: we want the fastest turn around on PR jobs, followed by nightly CI and dev jobs (since developers and contributors need to the see the latest state asap). Source deb jobs should happen before binary deb jobs and armhf is lowest priority (also taking the longest).

---

Each category contains jobs for each ROS 2 distro - atm 4 active distros. Newer distros have a higher priority than older distros (the rational being that they have more changes and more developers are interested in seeing them turnover).

The exception atm is `rolling` which has a lower priority than all other distros. As pointed out before I think this should be changed. E.g. the majority of PRs is targeting rolling and there the turn around for those should be of highest priority. The same argument holds true for the other categories - rolling is expected to change the most (same as comparing Foxy to Eloquent) and should therefore be handled before e.g. Foxy.

---

The `ros_buildfarm` package has hard coded jobs up to 35 and the upper limit is 100 which leaves 60ish priority levels to choose from. With 5 (or 6 categories if you count the architecture specific binary deb jobs separately) that still leave 10 priority levels for each category.

---

Therefore I am proposing the following changes:
* each category get a block of 10 per category (e.g. 40-49, 50-59, etc.) and
* in each category:
  * `rolling` gets `x0`
  * `foxy` get `x7`
  * `eloquent` get `x8`
  * `dashing` get `x9`

That will leave 6 priorities empty for future ROS 2 distros without collisions. Based on the current plan of releasing a new ROS 2 distro once a year that should give us plenty of room.

**Comment by jacobperron** [comment]
@dirk-thomas Your proposal SGTM :+1: 

What blocks do you think the job categories should get? Maybe leave available blocks at each end, e.g.

PR (2x)
CI (3x)
dev (4x)
source deb (5x)
binary deb amd64 / aarch64 (6x)
binary deb armhf (7x)

**Comment by dirk-thomas** [comment]
> What blocks do you think the job categories should get?

> PR (2x)

The "free" range only starts at 40 so I would start there which will go all the way to 9x. See #110 for the actual numbers (before and after).
